proxy/src/control/discovery.rs

```diff
@@ -162,44 +188,84 @@ impl Discovery {
 
         Watch {
             rx,
+            metric_labels: HashMap::new(),
             bind,
         }
     }
 }
 
 // ==== impl Watch =====
 
-impl<B> Discover for Watch<B>
+impl<B> Watch<B> {
+    fn update_metadata(&mut self,
+                       addr: SocketAddr,
+                       meta: Metadata)
+                       -> Result<(), ()>
+    {
+        if let Some(store) = self.metric_labels.get_mut(&addr) {
+            store.store(meta.metric_labels)
+                .map_err(|e| {
+                    error!("update_metadata: label store error: {:?}", e);
+                })
+                .map(|_| ())
+        } else {
+            // The store has already been removed, so nobody cares about
+            // the metadata change.
```

Comment:
As far as I can tell, this shouldn't happen; however, I thought it was better to handle this case gracefully instead of just making it `unreachable!()`; it's possible there's some edge case where this might happen.

Reply:
OK. To be clear, I think the implementation is right (like I said). I was just curious if we expect this to be dead code, to make sure I understand it.

Reply:
Yeah. I should probably change the comment to clear that up.